Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -79,11 +79,38 @@ public void DeserializesObjectsWithIgnoreDefaults()
             var serializer = new IonSerializer(new IonSerializationOptions {IgnoreDefaults = true});
             var deserialized = serializer.Deserialize<Motorcycle>(stream);
             
-            Assert.IsNull(deserialized.Brand);
+            Assert.IsNull(deserialized.Make);
             Assert.IsNull(deserialized.color);
             Assert.IsNotNull(deserialized.canOffroad);
         }
 
+        [TestMethod]
+        public void SerializesAndDeserializesObjectsWithCaseInsensitiveProperties()
+        {
+            var serializer = new IonSerializer(new IonSerializationOptions { PropertyNameCaseInsensitive = true });
+            var stream = serializer.Serialize(TestObjects.Harley);
+
+            var deserialized1 = serializer.Deserialize<MOTORCYCLE>(stream);
```

Comment:
Why is the Type name casing being changed here? This changes relates only to property names, not type names.

Reply:
`MOTORCYCLE` is just the name of the motorcycle class with capitalized property/field names.
Likewise with the lower case class `motorcyle` and mixed case class `MoToRcYcLe`.

Reply:
Please use a more expressive class name to show what you're really getting here. Something like this `ADifferentMotercycleClassWithTheSameButDifferentlyCasedProperties`

Reply:
Addresed.